app/main.py

```diff
@@ -1,8 +1,34 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.spouse = None
+        self.add_to_people()
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def add_to_people(self) -> None:
+        Person.people[self.name] = self
```

Comment:
maybe it's better to add person to people dictionary inside __init__ method to make sure person is only added to dictionary when new instance is created.